src/displayUI.js

```diff
@@ -6,52 +6,64 @@ import { instanceOfMainClass } from './mainClass';
 // Display one Category element
 export function displayOneCategory(categoryElement) {
 
+  let searchContent = document.querySelector("#searchBar");
```

Comment:
Wrong, not read input, you already have in class filter state, so don't read from ui if (searchContent.value === '') {
you have in model that state of input.